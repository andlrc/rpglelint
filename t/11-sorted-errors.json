[
   {
      "column":8,
      "file":"t/11-sorted-errors.rpgle",
      "line":2,
      "message":"global declaration 'a' is not allowed",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/11-sorted-errors.rpgle",
      "line":2,
      "message":"data structure 'a' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":7,
      "file":"t/11-sorted-errors.rpgle",
      "line":5,
      "message":"global declaration 'a' is not allowed",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/11-sorted-errors.rpgle",
      "line":6,
      "message":"global declaration 'b' is not allowed",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/11-sorted-errors.rpgle",
      "line":6,
      "message":"data structure 'b' needs to be qualified",
      "severity":"warning"
   },
   {
      "column":3,
      "file":"t/11-sorted-errors.rpgle",
      "line":7,
      "message":"declaration of 'sub1' shadows a global declaration",
      "severity":"warning"
   },
   {
      "column":3,
      "file":"t/11-sorted-errors.rpgle",
      "line":3,
      "message":"shadowed declaration is here",
      "severity":"note"
   }
]
