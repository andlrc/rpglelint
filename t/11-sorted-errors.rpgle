// rpglelint: -Wqualified -Wshadow -Wglobal
dcl-ds a;
  sub1 int(10);
end-ds;
dcl-s a int(10);
dcl-ds b;
  sub1 int(10);
end-ds;
