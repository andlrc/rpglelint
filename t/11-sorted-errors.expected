t/11-sorted-errors.rpgle:2:8: warning: global declaration 'a' is not allowed [-Wglobal]
 dcl-ds a;
        ^
t/11-sorted-errors.rpgle:2:8: warning: data structure 'a' needs to be qualified [-Wqualified]
 dcl-ds a;
        ^
t/11-sorted-errors.rpgle:5:7: warning: global declaration 'a' is not allowed [-Wglobal]
 dcl-s a int(10);
       ^
t/11-sorted-errors.rpgle:6:8: warning: global declaration 'b' is not allowed [-Wglobal]
 dcl-ds b;
        ^
t/11-sorted-errors.rpgle:6:8: warning: data structure 'b' needs to be qualified [-Wqualified]
 dcl-ds b;
        ^
t/11-sorted-errors.rpgle:7:3: warning: declaration of 'sub1' shadows a global declaration [-Wshadow]
   sub1 int(10);
   ^
t/11-sorted-errors.rpgle:3:3: note: shadowed declaration is here
   sub1 int(10);
   ^
